python/tk_framework_desktopserver/shotgun/scripts/execute_command.py

```diff
@@ -12,15 +12,30 @@
 import cPickle
 import os
 import logging
+import base64
+import functools
 
 # Special, non-engine commands that we'll need to handle ourselves.
 CORE_INFO_COMMAND = "__core_info"
 UPGRADE_CHECK_COMMAND = "__upgrade_check"
 
-# We can't have a global logger instance, because we can't import sgtk
-# in the global scope, but we can go ahead and define the logger name
-# to be used throughout the script.
-LOGGER_NAME = "wss2.execute_command"
+class _Formatter(logging.Formatter):
+    """
+    Custom logging formatter that base64 encodes all log messages.
+    """
+    def format(self, *args, **kwargs):
+        """
+        Encodes log messages as base64. This allows us to collapse multiline
+        log messages into a single line of text. Before presenting the log
+        messages to a user, the caller of execute_command.py will be required
+        to decode the message. Every message is tag at its head with "SGTK:",
+        making output from a logger using this formatter easily identifiable.
+        """
+        result = super(_Formatter, self).format(*args, **kwargs)
+        return "{tag}:{log_data}".format(
```

Comment:
minor/nitpick: why use format here when we use % style formatting everywhere else? seems inconsequential.